src/core/mesh/qgsmeshdataprovidertemporalcapabilities.h

```diff
@@ -0,0 +1,139 @@
+{
+  public:
+    /**
+     * Constructor for QgsMeshDataProviderTemporalCapabilities
+     */
+    QgsMeshDataProviderTemporalCapabilities();
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in milliseconds)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndexTimeInMilliseconds( int group, qint64 timeSinceGlobalReference ) const;
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in hours)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndex( int group, double  timeSinceGlobalReferenceInHours ) const;
+
+    /**
+     * Adds a \a reference date/time from a dataset \a group
+     *
+     * \note must be used only by the mesh data provider
+     */
+    void addGroupReferenceDateTime( int group, const QDateTime &reference ) SIP_SKIP;
+
+    /**
+     * Adds a \a time (in miliseonds) from a dataset contained in \a group
+     *
+     * \note must be used only by the mesh data provider,
+     * all dataset need to be added one after one
+     */
+    void addDatasetTimeInMilliseconds( int group, qint64 time ) SIP_SKIP;
+
+    /**
+     * Adds a \a time (in provider unit) from a dataset contained in \a group
+     *
+     * \note must be used only by the mesh data provider,
+     * all dataset need to be added one after one
+     */
+    void addDatasetTime( int group, double  time ) SIP_SKIP;
+
+    /**
+     * Returns true if the data has a reference time
+     *
+     */
+    bool hasReferenceTime() const;
+
+    /**
+     * Returns the reference time
+     *
+     */
+    QDateTime referenceTime() const;
+
+    /**
+     * Returns the time extent using the internal reference time
+     * and the first and last times available from the all the dataset
+     *
+     */
+    QgsDateTimeRange timeExtent() const;
+
+    /**
+     * Returns the time extent using \a reference date time
+     * and the first and last times available from the all the dataset
+     */
+    QgsDateTimeRange timeExtent( const QDateTime &reference ) const;
+
+    /**
+     * Sets the temporal unit (\a temporalUnit) used to read data
+     *
+     * Temporal units supported are milliseconds, seconds, minutes, hors, days and weeks
+     */
+    void setTemporalUnit( const QgsUnitTypes::TemporalUnit &temporalUnit );
+
+    /**
+     * Returns the temporal unit used to read data
+     *
+     * Temporal units supported are milliseconds, seconds, minutes, hors, days and weeks
+     *
+     */
+    QgsUnitTypes::TemporalUnit temporalUnit() const;
+
+    /**
+    * Returns the relative time in milliseconds of the dataset
+    *
+    */
+    qint64 datasetTime( const QgsMeshDatasetIndex &index ) const
```

Comment:
this function does not seem to me that it should be in the public api, maybe just copy code to the caller?

Reply:
This function is called by the List model class used to display the list of dataset for static dataset in the properties widget(https://github.com/vcloarec/QGIS/blob/6c6499f723d26d1a052ec12289ea95fd2023488e/src/app/mesh/qgsmeshstaticdatasetwidget.cpp#L112). 
So we need to expose it in the public API, no ?